draft-wang-ppm-dap-taskprov.md

```diff
@@ -451,11 +451,18 @@ out, it MUST not attempt to upload reports for the task.
 > OPEN ISSUE: In case of opt-out, would it be useful to specify how to report
 > this to the Author?
 
-Once the client opts into a task, it may begin uploading reports for the task.
-Each upload request for that task MUST advertise the task configuration. The
-extension codepoint `taskbind` MUST be offered in the `extensions` field of
-both Leader and Helper's `PlaintextInputShare`. In addition, each report's task
-ID MUST be computed as described in {{definition}}.
+Once the client opts into a task, it may begin uploading reports for the task
+to the Leader. The extension codepoint `taskbind` MUST be offered in the
+`extensions` field of both Leader and Helper's `PlaintextInputShare`. In
+addition, each report's task ID MUST be computed as described in {{definition}}.
+
+The Client MAY advertise the task configuration by specifying the encoded
+`TaskConfig` {{definition}} in the "dap-taskprov" HTTP header, but may choose
+to just specify the `taskbind` codepoint in the header, in order to save
+network bandwidth. However, the Leader may respond with "unrecognizedTask" if
```

Comment:
Hmmm, this is interesting. Why does client need to send the header at all? What do you expect the leader if it has opted in to the task, but the client doesn't send the header?

Reply:
The goal is to distinguish between taskprov and non-taskprov tasks. If we don't do this, I guess we can decrypt leader share and try to find the extension codepoint. Both should work. Do you prefer the latter?

Reply:
If the leader has opted in, then it knows that the task is taskprov. I don't think signaling in the header as well as necessary. Dropping it here would make the protocol simpler, so I'd be in favor of dropping it.

Reply:
Sounds good. Updated the PR.